docs/references/ayx-plugin-cli.md

```diff
@@ -0,0 +1,487 @@
+half-finished files and directories in the workspace. In order to delete
+these, follow these steps:
+
+-   Remove config/\<Plugin\> directory.
+-   Remove backend/ayx_plugins/\<plugin\>.py.
+-   Force-remove ui/\<Plugin\>.
+-   Remove the import statement from \_\_init\_\_.py.
+-   Remove the tool from ayx_workspace.json's "tools" key.
+
+### Parameters
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
+
+#### \--tool-name
+
+This parameter becomes the name of the tool. Spaces are allowed but will
+be cleaned into underscores and Pascal case. For example, for a Python
+plugin named "A Test Tool", the filename is `a_test_tool.py`, and the
+`class/ui/config` name is `ATestTool`.
+
+-   Optional or Required: Required
+-   Type: String
+-   Permissible Values: Any alphanumeric string that doesn't start with
+    a number and doesn't shadow reserved names.
+
+#### \--tool-type
+
+This parameter is used to determine the scaffold that the tool should
+generate for a plugin.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: input, output, multiple-inputs,
+    multiple-outputs, single-input-single-output
+-   Default: single-input-single-output
+
+#### \--description
+
+The plugin description.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+#### \--version
+
+The plugin version.
+
+-   Optional or Required: Optional
+-   Type: String
+-   Permissible Values: Any
+
+## create-yxi
+
+This command packages a workspace into a YXI that can be installed into
+Designer. The resulting YXI is located under `build/yxi/`.
+
+### Parameters
+
+This command takes no parameters.
+
+## install-yxi
+
+This command takes a YXI (any CLI-built or legacy SDK YXI) and installs
+it into Designer.
+
+Note: This command can be run from anywhere, not just Alteryx Workspace
+directories.
+
+### Parameters
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
+
+#### \--yxi-path
+
+The location of the YXI to be installed. Note that this does not work
+for YXIs built for the original Alteryx Python SDK.
+
+-   Optional or Required: Required
+-   Type: String
+-   Permissible Values: Any valid path to a YXI.
+
+#### \--install-type
+
+This parameter is used to determine where a plugin should be installed.
+Note that admin installation requires admin access.
+
+-   Optional or Required: Required
+-   Type: String
+-   Permissible Values: user, admin
+-   Default: user
+
+## designer-install
+
+This command builds the YXI from the current workspace and installs it
+into Designer.
+
+### Parameters
+
+Note: If a parameter is not passed in as part of the CLI call, the CLI
+will wait for standard input for the ignored parameters, even if they're
+optional.
```

Comment:
TW review:

Note: If a parameter is not passed in as part of the CLI call, the CLI
waits for standard input for the ignored parameters, even if they're
optional.

Reply:
fixed